overlord/snapstate/flags.go

```diff
@@ -65,6 +65,9 @@ type Flags struct {
 	// re-refresh tasks. This allows refresh to work offline, as
 	// long as refresh assets are cached.
 	NoReRefresh bool `json:"no-rerefresh,omitempty"`
+
+	// RequireBaseType is set to mark that a snap needs to be of type: base, otherwise installation fails.
+	RequireBaseType bool `json:"require-base-type"`
```

Comment:
I think we want to call this RequireTypeBase (and the same for all the other places s/BaseType/TypeBase/)

Reply:
We should add code to Flags.ForSnapSetup (in flags.go) to reset this, also it should be  "omitempty"

Reply:
Done.